torch/_dynamo/variables/tensor.py

```diff
@@ -391,16 +391,21 @@ def method_attr__version(self, tx):
         )
 
     def call_obj_hasattr(self, tx: "InstructionTranslator", name):
-        from . import GetAttrVariable
-        from .builtin import BuiltinVariable
+        if tx.output.side_effects.has_pending_mutation_of_attr(self, name):
+            return ConstantVariable(True)
 
         try:
-            var = BuiltinVariable(getattr).call_function(
-                tx, [self, ConstantVariable(name)], {}
+            self.var_getattr(tx, name)
+            ret_val = True
+        except NotImplementedError:
+            unimplemented_v2(
```

Comment:
Should you also pass the from arg here with the NotImplementedError to properly chain the exception?